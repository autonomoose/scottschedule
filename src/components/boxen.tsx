/*
    various boxes
*/
import React from 'react';
import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';

interface iCaptionBox {
    caption: string,
    color?: string,
    xpad?: string, // rem
}
// quick caption box
export const CaptionBox = (props: iCaptionBox) => {
    const { caption, color, xpad } = props;
    const wkColor = color || 'inherit';
    const wkXpad = xpad || '0.5rem';
    return (
      <Box px={wkXpad} mt={-1.75}>
        <Typography variant='caption' color={wkColor}
          sx={{ bgcolor:'background.paper', position: 'relative', zIndex: '2'}} >
          &nbsp;{caption}&nbsp;
        </Typography>
      </Box>
)}

