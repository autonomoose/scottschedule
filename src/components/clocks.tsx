// more clocks
//   - clocks must avoid hooks, clocks are conditionals
//   - define new clock and register it with widget

import React from 'react';

import Box from '@mui/material/Box';
import Button from '@mui/material/Button';
import Typography from '@mui/material/Typography';

// use this as an outside interface to clocks
interface ClockWidgetProps {
  currClock: string,
  onComplete: (status: string) => void,
  currSched?: string,
  descrSched?: string,
}
export const ClockWidget = (props: ClockWidgetProps) => {
    return(
      <Box display='flex' flexDirection='column'>
      {
        {
          'digital1': <ClockDigital1 onComplete={props.onComplete} />,
          'digital2': <ClockDigital2 onComplete={props.onComplete} />,
        }[props.currClock]
      }
      </Box>
)}

interface ClockProps {
  onComplete: (status: string) => void,
}
export const ClockDigital2 = (props: ClockProps) => {
    return(
      <>
        <Button data-testid='change clock' sx={{margin: 0}} onClick={() => props.onComplete('scheduler')}>
          <Typography variant='h1' id='mainclock' data-testid='mainclock'
            sx={{fontSize:150, fontWeight: 600, padding: 0, margin: 0}}>00:00</Typography>
        </Button>
        <Typography variant='subtitle1' id='mainpm' data-testid='mainpm' sx={{fontSize:20, fontWeight: 600}}>
          PM
        </Typography>
        <Box mx={4} display='flex' justifyContent='space-between'>
          <Typography mx={1} variant='h4' id='maindate' data-testid='maindate'>Day, 01/01</Typography>
          <Button onClick={() => props.onComplete('close')}>Scheduler</Button>
        </Box>
      </>
) };

export const ClockDigital1 = (props: ClockProps) => {
    return(
      <>
        <Button data-testid='change clock' sx={{margin: 0, padding: 0}} onClick={() => props.onComplete('digital2')}>
          <Box display='flex' justifyContent='flex-start' alignItems='stretch'>
            <Box mt={1} display='flex' alignItems='flex-start'>
              <Typography id='mainpm' data-testid='mainpm'
               sx={{lineHeight:1, fontSize:20, fontWeight: 600}}>
                PM
              </Typography>
            </Box>
            <Typography id='compclock' data-testid='compclock'
             sx={{marginLeft: '-2px', letterSpacing: '-30px', lineHeight:.8, fontSize:230, fontWeight: 600, padding: 0, margin: 0}}>
               0
            </Typography>
            <Box ml={3} mt={1}>
              <Typography id='compminutes' data-testid='compminutes'
               sx={{letterSpacing: '-4px',lineHeight:.92, fontSize:150, fontWeight: 800, padding: 0, margin: 0}}>
                00
              </Typography>
              <Typography id='maindate' data-testid='maindate'
               sx={{ textTransform: 'none', lineHeight:1, fontSize:26, padding: 0, margin: 0}}>
                Day, 01/01
              </Typography>
            </Box>
           </Box>
        </Button>
        <Box mx={4} display='flex' justifyContent='flex-end'>
          <Button onClick={() => props.onComplete('digital1-color')}>Color</Button>
          <Button onClick={() => props.onComplete('close')}>Scheduler</Button>
        </Box>
      </>
) };



