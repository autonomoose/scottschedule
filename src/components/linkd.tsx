// this rather convoluted hack is a compromise
//  between ui link, gatsby link, and typescript
import React from 'react';
import type { GatsbyLinkProps } from "gatsby";
import { Link } from 'gatsby';

import MuiLink from '@mui/material/Link';

/*
    Use this to wrap gatsby link in mui components
    such as button, etc.  link example <LinkD> below
        component={}
*/

export type MyLinkProps = Omit<GatsbyLinkProps<unknown>, "ref">;
export const GatsbyLink = React.forwardRef<any, MyLinkProps>((props, ref) => (
    <Link
      innerRef={ref} activeClassName="active"
      {...props}
    />
),);

/*
  LinkD demonstrates the embeddable gatsby link and extends it
    to pass an explicit debug= on the URL when passed as a parm
    vdebug
*/
const linkDebug = (plink: string, vdebug: string): string => {
    if (!vdebug) { return(plink); }
    const sep = (plink.indexOf('?') > 0) ? '&': '?';
    return (plink + sep + 'debug=' + vdebug);
}
export const LinkD = (props: any) => {
    const { to, vdebug, children, ...pass_props } = props;
    return (
        <MuiLink component={GatsbyLink} to={linkDebug(to, vdebug)} {...pass_props}>{children}</MuiLink>
    );
}

export default LinkD

