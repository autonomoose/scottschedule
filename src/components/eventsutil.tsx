// events utilities and components
// exports default DisplayEvents,
//  also exports components CreateEvent, ModifyEvent
//  and data fetchEventsDB - full events
import React, { useEffect, useState } from 'react';
import { API } from 'aws-amplify';
import { useForm } from "react-hook-form";
import { ErrorMessage } from '@hookform/error-message';

import {CaptionBox} from './boxen';

import Box from '@mui/material/Box';
import Button from '@mui/material/Button';
import Card from '@mui/material/Card';
import Grid from '@mui/material/Grid';
import IconButton from '@mui/material/IconButton';
import List from '@mui/material/List';
import ListItem from '@mui/material/ListItem';
import TextField from '@mui/material/TextField';
import Typography from '@mui/material/Typography';

import DeleteForeverIcon from '@mui/icons-material/DeleteForever';

import { listEventsFull, iTaskDB } from '../graphql/queries';
import { mutAddEvents, mutAddRules, mutDelEvents } from '../graphql/mutations';

// -------------------------------------------------
interface CreateRuleProps {
  evName: string,
  onComplete: (status: string) => void,
  open: boolean
}
export const CreateRule = (props: CreateRuleProps) => {
    // form states
    const { register, handleSubmit, reset, formState, watch } = useForm({
        defaultValues: {
            cmd: 'begin',
            options: '',
            content: '',
        }
    });
    const { isDirty, errors } = formState;

    const formNewRuleCancel = async () => {
        props.onComplete('');
    }
    interface FormNewRuleParms {
        cmd: string,
        options: string,
        content: string,
    };
    const formNewRuleSubmit = async (data: FormNewRuleParms) => {
        try {
            let wkEvNames = props.evName+"!"+data.cmd;
            if (data.options !== '') {
                wkEvNames += ' ' + data.options;
            }
            const xdata = {'input': {
                'etype': 'ev',
                'evnames': wkEvNames,
                'rules': data.content,
                }
            };
            await API.graphql({query: mutAddRules, variables: xdata});
            props.onComplete(props.evName);

        } catch (result) {
            console.warn('failed update rules', result);
            props.onComplete('');
        }
    };

    return(
      <Box display={(props.open)?'block': 'none'}>
      <Card style={{marginTop: '3px', maxWidth: 350, minWidth: 350, flex: '1 1',
       boxShadow: '-5px 5px 12px #888888', borderRadius: '0 0 5px 5px'}}>
        <Box mx={1}>
          <form key="newRule" onSubmit={handleSubmit(formNewRuleSubmit)}>
          <Box display="flex" justifyContent="space-between" alignItems="baseline">
            <Typography variant='h6'>
              Add New Rule
            </Typography>
          </Box>

          <Box display='flex'>
            <select {...register("cmd")}>
              <option value="begin">begin</option>
              <option value="option">option</option>
            </select>
            <input disabled={(watch('cmd') === 'begin')} data-testid="optionsInput" {...register('options')} />
          </Box>

          <Box>
            <textarea rows={2} cols={27} data-testid="contentRule"
             {...register('content', { required: true,})}
             aria-invalid={errors.content ? "true" : "false"}
            ></textarea>
          </Box>

          <Box mt={2} display='flex' justifyContent='flex-end'>
            <Button size="small" variant="outlined" color='error' onClick={() => formNewRuleCancel()}>Cancel</Button>
            <Button size="small" variant="outlined" onClick={() => reset()} disabled={!isDirty}>Reset</Button>
            <Button size="small" variant="contained" type="submit" disabled={!isDirty}>Save</Button>
          </Box>

          </form>
        </Box>
      </Card></Box>
) };

/*
   modify events
*/
interface ModifyEventProps {
    evid: string,
    tasks: iTask,
    onComplete: (status: string) => void,
    open: boolean,
}
interface FormModifyEventParms {
    evName: string,
    descr: string,
    sound: string,
    soundrepeat: string,
}
export const ModifyEvent = (props: ModifyEventProps) => {
    const allTasks = props.tasks;
    const evid = props.evid;

    const [evRule, setEvRule] = useState('');

    // form states
    const formDefaultVal: FormModifyEventParms = {
        evName: '',
        descr: '',
        sound: '_default_',
        soundrepeat: '0',
    };
    const { register, handleSubmit, reset, formState } = useForm({defaultValues: formDefaultVal});
    const { isDirty, errors } = formState;

    /* -------- reset form defaults to any current values ------ */
    useEffect(() => {
        let defaultValues: FormModifyEventParms = formDefaultVal;
        if ( evid && evid !== '_new' ) {
            const wkev = allTasks[evid];
            // modify default vals for existing event
            defaultValues['evName'] = evid;
            if (wkev?.descr) {
                defaultValues['descr'] = wkev.descr;
            }
            if (wkev?.sound) {
                if (wkev.sound['name'] || wkev.sound['name'] === '') {
                    defaultValues['sound'] = wkev.sound.name;
                }
                if (wkev.sound['repeat']) {
                    defaultValues['soundrepeat'] = wkev.sound.repeat.toString();
                }
            }
        }

        reset(defaultValues);
    }, [allTasks, evid] );

    // success form submit, map the form data onto the graphql input
    const formModEvSubmit = async (data: FormModifyEventParms) => {
        try {
            const xdata = {'input': {
                'etype': 'ev',
                'evnames': data.evName+"!args",
                'descr': data.descr,
                'sound': data.sound,
                'soundrepeat': data.soundrepeat,
                }
            };
            await API.graphql({query: mutAddEvents, variables: xdata});
            props.onComplete(data.evName);
        } catch (result) {
            console.warn('failed update', result);
        }
    };
    interface FormDelEventParms {
        cmd: string,
    };
    const formDelEvent = async (data: FormDelEventParms) => {
        try {
            const xdata = {'input': {
                'etype': 'ev',
                'evnames': evid+"!"+data.cmd,
                }
            };
            await API.graphql({query: mutDelEvents, variables: xdata});
            props.onComplete(evid);
        } catch (result) {
            console.warn('failed delete', result);
        }
    };
    const formCallback = (status: string) => {
        console.log("mod callback status", status);
        setEvRule('');
        props.onComplete(status);
    };

    return(
      <Box ml={4} display={(props.open)?'block': 'none'}>
      <Card style={{margin: '3px 0 0 0', maxWidth: 350, minWidth: 350, flex: '1 1',
       boxShadow: '-5px 5px 12px #888888', borderRadius: '0 0 5px 5px'}}>
        <Box>
          <form key="newEv" onSubmit={handleSubmit(formModEvSubmit)}>
          {/* -------------- Title block ----------------- */}
          <Box px='0.5em' display="flex" justifyContent="space-between" alignItems="baseline" sx={{bgcolor: 'site.main'}}>
            <Typography variant='h6'>
              { (evid && evid !== '_new') ?
                <span>Event {evid}
                  { (allTasks[evid] && allTasks[evid].schedRules.length === 0) ?
                    <IconButton data-testid={'del-'+evid} size='small' color='error' onClick={() => formDelEvent({'cmd': 'args'})}>
                      <DeleteForeverIcon sx={{height: '1.25rem'}} />
                    </IconButton>
                    :
                    <IconButton disabled data-testid={'del-'+evid} size='small'><DeleteForeverIcon sx={{height: '1.25rem'}} /></IconButton>
                  }
                </span>
                :
                <span>
                  Add New Event
                </span>
              }
            </Typography>
            <IconButton data-testid='cancel' size='small' onClick={() => props.onComplete('')}>X</IconButton>
          </Box>

          {/* -------------- Main Form Grid ----------------- */}
          <Box sx={{ flexGrow: 1}}><Grid container spacing={2}>
            {/* -------------- Top Line ----------------- */}
            {/* -------------- for Add this is a text box ----------------- */}
            <Grid item xs={5}>
              <Box px='0.5rem' mt={.5} display={(evid && evid !== '_new')?'none':'block'}>
                <TextField label='Name' size='small'
                  {...register('evName', {
                    required: 'this field is required',
                    pattern: {value: /^[a-zA-Z0-9]+$/, message: 'alphanumeric only'},
                    maxLength: {value: 12, message: 'limited to 12 characters'},
                  })}
                  color={errors.evName ? 'error' : 'primary'}
                  aria-invalid={errors.evName ? "true" : "false"}
                  inputProps={{'data-testid': 'nameInput'}}
                  InputLabelProps={{shrink: true}}
                />
                <ErrorMessage errors={errors} name="evName" render={({ message }) =>
                  <CaptionBox caption={message} color='error'/>
                } />
              </Box>
            </Grid>

            {/* -------------- summary ------ */}
            <Grid item xs={7}>
              <Box mt={.5} mr={.5} px='0.5rem'>
                <TextField label='Summary' size='small' fullWidth
                  {...register('descr', {
                    required: 'this field is required',
                    pattern: {value: /^[a-zA-Z0-9 \-]+$/, message: 'no special characters'},
                    maxLength: {value: 20, message: 'limited to 20 characters'},
                  })}
                  aria-invalid={errors.descr ? "true" : "false"}
                  color={errors.descr ? 'error' : 'primary'}
                  inputProps={{'data-testid': 'descrInput'}}
                  InputLabelProps={{shrink: true}}
                />
                <ErrorMessage errors={errors} name="descr" render={({ message }) =>
                  <CaptionBox caption={message} color='error'/>
                } />
              </Box>
            </Grid>

            {/* -------------- sound ------ */}
            <Grid item xs={12} display='flex' my={2}>
              {/* -------------- name ------ */}
              <Grid item xs={6} >
                <Box px='0.5rem' >
                  <TextField label='Sound' size='small' fullWidth
                    {...register('sound', {
                      pattern: {value: /^[a-zA-Z0-9\-\_]+$/, message: 'no special chars'},
                      maxLength: {value: 20, message: 'too long - limited to 10 characters'},
                    })}
                    aria-invalid={errors.sound ? "true" : "false"}
                    color={errors.sound ? 'error' : 'primary'}
                    inputProps={{'data-testid': 'soundInput'}}
                  />
                </Box>
                <Box px={1.5}>
                  <ErrorMessage errors={errors} name="sound" render={({ message }) =>
                    <CaptionBox caption={message} color='error'/>
                  } />
                </Box>
              </Grid> {/* end sound name */}

              {/* -------------- sound repeat ------ */}
              <Grid item xs={3} >
                <Box px='0.5rem' >
                  <TextField label='Repeat' size='small' fullWidth
                    {...register('soundrepeat', {
                      pattern: {value: /^[0-9]+$/, message: 'numeric only'},
                      maxLength: {value: 2, message: 'less than 100'},
                    })}
                    aria-invalid={errors.soundrepeat ? "true" : "false"}
                    inputProps={{'data-testid': 'soundRepeatInput'}}
                  />
                </Box>
                <Box px={1.5}>
                  <ErrorMessage errors={errors} name="soundrepeat" render={({ message }) =>
                    <CaptionBox caption={message} color='error'/>
                  } />
                </Box>
              </Grid> {/* end soundrepeat */}
            </Grid> {/* end sound */}

            {/* ---------- Future fields! ----------------- */}

          </Grid></Box> {/* ----------End Grid ----------------- */}

          {/* ---------- Form Save/Reset Buttons ----------------- */}
          <Box px='0.5em' mt={2} display='flex' justifyContent='flex-end'>
            <Button size="small" variant="outlined" onClick={() => reset()} disabled={!isDirty}>Reset</Button>
            <Button size="small" variant="contained" color="primary" type="submit" disabled={!isDirty}>Save</Button>
          </Box>
          </form>

          {/* -------------- Rules ----------------- */}
          <Box px='0.5em'  mt={2} mb={1} display='flex' justifyContent='space-between' sx={{bgcolor: 'site.main'}}>
            <span>Rules ({(allTasks && allTasks[evid])? allTasks[evid].schedRules.length: 0})</span>
            <Button onClick={() => setEvRule(evid)}  size="small" variant="outlined" color="primary">New Rule</Button>
          </Box>
          {(allTasks && allTasks[evid]) &&
          <Box>
            <CreateRule evName={evRule} onComplete={formCallback} open={(evRule !== '')} />
            {
              allTasks[evid].schedRules.map((task: string) => {
                // get the cmd words out
                let words = task.split(' ');
                let cmd = words.shift() || 'begin';
                if (cmd === 'option') {
                    cmd = 'option ' +  words.shift() || '';
                }
                const displayRule = words.join(' ').split(',').join(', ');

              return(
                <Box key={`${evid}${cmd}`}  display='flex'>
                  <Box display='flex'>
                    <IconButton size='small' color='error' onClick={() => formDelEvent({cmd})}>X</IconButton>
                    {cmd}
                  </Box>
                  <Box mx={1} mb={1} px={1} sx={{ border: '1px solid grey' }} key={`${evid}${cmd}`}>
                    {displayRule}
                  </Box>
                </Box>
              )})
            }

          </Box>
          }

        </Box>
      </Card></Box>
) };

// -------------------------------------------------
interface DisplayEventProps {
  tasks: iTask,
  select?: (evid: string) => void,
}
const DisplayEvents = (props: DisplayEventProps) => {
    const allTasks = props.tasks;

    return(
      <List disablePadding dense sx={{marginLeft: '1em'}}>
        {
          Object.keys(allTasks).map((evid: string) => {
          return(
            <ListItem button key={evid} onClick={() => {props.select?.(evid);}}>
              {evid} - {allTasks[evid].descr}
            </ListItem>
           )})
        }
      </List>
) };

// -------------------------------------------------
export const fetchEventsDB = async (): Promise<iTask> => {
    try {
        const result: any = await API.graphql({query: listEventsFull})

        const compactTasks = result.data.listEvents.items.reduce((resdict: iTask, item: iTaskDB) => {
            const evkeys = item.evnames.split('!');
            if (!resdict[evkeys[0]]) {
                resdict[evkeys[0]] = {descr: '', schedRules: []};
            }
            if (evkeys[1] === 'args') {
                let wkSound: iEvsSound = {};
                resdict[evkeys[0]].descr = (item.descr)? item.descr: '';
                if (item.sound || item.sound === '') {
                    if (item.sound !== '_default_') {
                        wkSound['name'] = item.sound;
                    }
                }
                if (item.soundrepeat && parseInt(item.soundrepeat, 10) !== 0) {
                    wkSound['repeat'] = parseInt(item.soundrepeat, 10);
                }
                if (wkSound && Object.keys(wkSound).length > 0) {
                    resdict[evkeys[0]].sound = wkSound;
                }
            } else {
                resdict[evkeys[0]].schedRules.push(evkeys[1] + " " + item.rules);
            }
            return resdict;
        }, {});

        return(compactTasks);
    } catch (result) {
        return({});
    }
};

export default DisplayEvents
