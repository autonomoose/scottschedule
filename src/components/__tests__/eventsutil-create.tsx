import React from "react";
import { render, waitFor } from "@testing-library/react";
import userEvent from '@testing-library/user-event'

import { API } from 'aws-amplify';
import { ModifyEvent } from '../../components/eventsutil';
jest.mock('aws-amplify');

const mockCallback = jest.fn();
const mockEvents = {testevt: {descr: 'testing', schedRules: ["begin +2,++2,++2"]}};

const mytest = <ModifyEvent evid='_new' tasks={mockEvents} onComplete={mockCallback} open={true} />;
const mySetup = () => {
    const utils = render(mytest);
    const resetButton = utils.getByRole('button', {name: /reset/i});
    const saveButton = utils.getByRole('button', {name: /save/i});
    const nameFld = utils.getByTestId('nameInput');
    const descrFld = utils.getByTestId('descrInput');

    return {
        ...utils,
        resetButton,
        saveButton,
        nameFld,
        descrFld,
    }
};

describe("eventsutil - create", () => {
  it("renders snapshot correctly", () => {
    const {container} = render(mytest);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("starts with buttons in correct status", () => {
    const utils = mySetup();

    expect(utils.resetButton).toBeDisabled();
    expect(utils.saveButton).toBeDisabled();
  });
  it("enables reset and save after name modification", async () => {
    const utils = mySetup();

    userEvent.type(utils.nameFld, 'newgrp');
    await waitFor(() => {
      expect(utils.resetButton).toBeEnabled();
    });
    expect(utils.saveButton).toBeEnabled();
  });
  it("handles reset after name modification", async () => {
    const utils = mySetup();

    userEvent.type(utils.nameFld, 'newgrp');
    await waitFor(() => {
      expect(utils.resetButton).toBeEnabled();
    });
    userEvent.click(utils.resetButton);
    await waitFor(() => {
      expect(utils.resetButton).toBeDisabled();
    });
  });
  it("handles graphql error on save", async () => {
    const consoleWarnFn = jest.spyOn(console, 'warn').mockImplementation(() => jest.fn());
    const prevAPIgraphql = API.graphql;
    API.graphql = jest.fn(() => Promise.reject('mockreject')) as any;
    const utils = mySetup();

    userEvent.type(utils.nameFld, 'newgrp');
    userEvent.type(utils.descrFld, 'new desc');
    await waitFor(() => {
      expect(utils.resetButton).toBeEnabled();
    });
    expect(utils.saveButton).toBeEnabled();
    userEvent.click(utils.saveButton);

    await waitFor(() => {
      expect(consoleWarnFn).toHaveBeenCalledTimes(1);
    });
    API.graphql = prevAPIgraphql;
    consoleWarnFn.mockRestore();
  });
  it("handles save after name modification", async () => {
    const prevAPIgraphql = API.graphql;
    API.graphql = jest.fn(() => Promise.resolve({})) as any;
    const utils = mySetup();

    userEvent.type(utils.nameFld, 'newgrp');
    userEvent.type(utils.descrFld, 'new desc');
    await waitFor(() => {
      expect(utils.resetButton).toBeEnabled();
    });
    expect(utils.saveButton).toBeEnabled();
    userEvent.click(utils.saveButton);

    await waitFor(() => {
      expect(mockCallback).toHaveBeenLastCalledWith('newgrp');
    });
    API.graphql = prevAPIgraphql;
  });

});


