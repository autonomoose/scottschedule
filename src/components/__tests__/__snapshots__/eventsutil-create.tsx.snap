// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eventsutil - create renders snapshot correctly 1`] = `
<div
  class="MuiBox-root css-2t3rs8"
>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    style="margin: 3px 0px 0px 0px; max-width: 350px; min-width: 350px; flex: 1 1; box-shadow: -5px 5px 12px #888888; border-radius: 0 0 5px 5px;"
  >
    <div
      class="MuiBox-root css-0"
    >
      <form>
        <div
          class="MuiBox-root css-1j6jckw"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            <span>
              Add New Event
            </span>
          </h6>
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
            data-testid="cancel"
            tabindex="0"
            type="button"
          >
            X
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiBox-root css-i9gxme"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
            >
              <div
                class="MuiBox-root css-hzaddc"
              >
                <div
                  aria-invalid="false"
                  class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="true"
                    for="mui-1"
                    id="mui-1-label"
                  >
                    Name
                  </label>
                  <div
                    class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                      data-testid="nameInput"
                      id="mui-1"
                      name="evName"
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-1in441m"
                      >
                        <span>
                          Name
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7 css-uqwprf-MuiGrid-root"
            >
              <div
                class="MuiBox-root css-150uz6c"
              >
                <div
                  aria-invalid="false"
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="true"
                    for="mui-2"
                    id="mui-2-label"
                  >
                    Summary
                  </label>
                  <div
                    class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                      data-testid="descrInput"
                      id="mui-2"
                      name="descr"
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-1in441m"
                      >
                        <span>
                          Summary
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-41ti7o-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
              >
                <div
                  class="MuiBox-root css-1ozu5o9"
                >
                  <div
                    aria-invalid="false"
                    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                  >
                    <label
                      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
                      data-shrink="true"
                      for="mui-3"
                      id="mui-3-label"
                    >
                      Sound
                    </label>
                    <div
                      class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                        data-testid="soundInput"
                        id="mui-3"
                        name="sound"
                        type="text"
                        value=""
                      />
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-1in441m"
                        >
                          <span>
                            Sound
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiBox-root css-hhy13k"
                />
              </div>
               
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
              >
                <div
                  class="MuiBox-root css-1ozu5o9"
                >
                  <div
                    aria-invalid="false"
                    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                  >
                    <label
                      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
                      data-shrink="true"
                      for="mui-4"
                      id="mui-4-label"
                    >
                      Repeat
                    </label>
                    <div
                      class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                        data-testid="soundRepeatInput"
                        id="mui-4"
                        name="soundrepeat"
                        type="text"
                        value=""
                      />
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-1in441m"
                        >
                          <span>
                            Repeat
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiBox-root css-hhy13k"
                />
              </div>
               
            </div>
             
          </div>
        </div>
         
        <div
          class="MuiBox-root css-1m55zth"
        >
          <button
            class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root Mui-disabled  css-1k23hlb-MuiButtonBase-root-MuiButton-root"
            disabled=""
            tabindex="-1"
            type="button"
          >
            Reset
          </button>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButtonBase-root Mui-disabled  css-11qr2p8-MuiButtonBase-root-MuiButton-root"
            disabled=""
            tabindex="-1"
            type="submit"
          >
            Save
          </button>
        </div>
      </form>
      <div
        class="MuiBox-root css-brm5ry"
      >
        <span>
          Rules (
          0
          )
        </span>
        <button
          class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root  css-1k23hlb-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          New Rule
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
